### PR TITLE
Milight

### PR DESCRIPTION
**Pull request by Lyr3x**
@Nesurion lgty?

### CODE REVIEW

File: addons/milight.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    socket = UDPSocket.new
+    socket.send(SPEED_DOWN, 0, WIFI_BRIDGE_IP, WIFI_BRIDGE_PORT)
+  end
+
+  def self.night()
+    socket = UDPSocket.new
+      t = Time.now
+      loop do
+        socket.send(LAMP_OFF, 0, WIFI_BRIDGE_IP, WIFI_BRIDGE_PORT)
+        break if Time.now > t
```
Comment (Nesurion) [inline addons/milight.rb]: what does this do? or what is the intention here?

File: addons/milight.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      white
+    when "disco"
+      disco
+    when "faster"
+      faster
+    when "slower"
+      slower
+    when "night"
+      night
+    else puts "wrong command"
```
Comment (Nesurion) [inline addons/milight.rb]: maybe `unknown command` easier to debug

File: addons/milight.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    socket = UDPSocket.new
+    socket.send(SPEED_UP, 0, WIFI_BRIDGE_IP, WIFI_BRIDGE_PORT)
+  end
+
+  def self.slower()
+    socket = UDPSocket.new
+    socket.send(SPEED_DOWN, 0, WIFI_BRIDGE_IP, WIFI_BRIDGE_PORT)
+  end
+
+  def self.night()
```
Comment (ZeissS) [inline addons/milight.rb]: where does this end? (there is no `end`)

File: addons/milight.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+module Milight
+  def self.config()
+      milight_config = File.expand_path(
+                       File.join(
+                         ENV['JANE_PATH'], 'config', 'milight.json'
+                       )
+                     )
+      return JSON.parse(File.read(milight_config), symbolize_names: true)
+    end
```
Comment (ZeissS) [inline addons/milight.rb]: Indentation!

File: addons/milight.rb (hunk 3 above)
Comment (Lyr3x) [inline addons/milight.rb]: already fixed

File: addons/milight.rb (hunk 4 above)
Comment (Lyr3x) [inline addons/milight.rb]: ill fix this later :+1:

File: addons/milight.rb [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+  #define speed states
+  SPEED_UP = "\x44\x00\x55"
+  SPEED_DOWN = "\x43\x00\x55"
+
+  #define night mode
+  NIGHT_MODE = "\xC6\x00\x55"
+
+  def self.on()
+    socket = UDPSocket.new
```
Comment (ZeissS) [inline addons/milight.rb]: you can make your life a lot easier by either caching this or putting it into a helper function:

```
def self.socket
   UDPSocket.new
end

def self.other 
    socket.send ...
end

```

Not sure my ruby is correct here, but you are repeating yourself over and over ;)

Comment (Lyr3x) [inline addons/milight.rb]: i see. ill change this

**Pull request merged**